src/client/opamState.ml

```diff
@@ -1970,7 +1970,7 @@ let update_ocamlinit () =
 let string_of_env_update t shell updates =
   let fenv = resolve_variable t OpamVariable.Map.empty in
   let sh   (k,v) = Printf.sprintf "%s=%S; export %s;\n" k v k in
-  let csh  (k,v) = Printf.sprintf "setenv %s %S;\n" k v in
+  let csh  (k,v) = Printf.sprintf "if ( ! ${?%s} ) setenv %s \"\"\nsetenv %s %S\n" k k k v in
```

Comment:
Why is this needed ? I am curious how it differs from the simple `setenv %s %S`...